test/e2e-go/features/accountPerf/sixMillion_test.go

```diff
@@ -1010,30 +1017,19 @@ func handleError(err error, message string, errChan chan<- error) {
 
 // handle the counters to prepare and send transactions in batches of MaxTxnLife transactions
 func checkPoint(counter, firstValid, tLife uint64, force bool, fixture *fixtures.RestClientFixture, log logging.Logger) (newCounter, nextFirstValid uint64, err error) {
-	waitBlock := 5
 	lastRound := firstValid + counter - 1
 	if force || counter == tLife {
 		if verbose {
 			fmt.Printf("Waiting for round %d...", int(lastRound))
 		}
-		for x := 0; x < 1000; x++ {
-			err := fixture.WaitForRound(lastRound, time.Duration(waitBlock)*time.Second)
-			if err == nil {
-				if verbose {
-					fmt.Printf(" waited less than %d sec, done.\n", (x+1)*waitBlock)
-				}
-				status, err := fixture.AlgodClient.Status()
-				if err != nil {
-					return 0, lastRound + 1, nil
-				}
-				return 0, status.LastRound + 1, nil
-			}
-			if verbose {
-				fmt.Printf(" waited %d sec, continue waiting...", (x+1)*waitBlock)
-			}
+		nodeStat, err := fixture.AlgodClient.WaitForBlock(basics.Round(lastRound - 1))
```

Comment:
This is great, but loosing the timeout and the ability to monitor the progress.

Reply:
I think that in general I agree with your points. But in this case, the entire test is geared toward success, so I don't think that this is an issue. The WaitForBlock would exit with an error after 60 seconds if it's unable to reach the block.